src/stores/pacbioPoolCreate.js

```diff
@@ -618,35 +626,32 @@ export const usePacbioPoolCreateStore = defineStore('pacbioPoolCreate', {
         } = groupIncludedByResource(included)
 
         //Populate pool attributes
-        this.pool = {
-          id: data.id,
-          ...data.attributes,
-        }
+        this.pool = extractAttributes(data)
 
         //Populate requests
         this.resources.requests = dataToObjectById({
           data: requests,
           includeRelationships: true,
         })
-        //Populate tubes
-        this.resources.tubes = dataToObjectById({ data: tubes, includeRelationships: true })
-
-        // Get the pool tube and remove it from tubes list
-        this.tube = this.resources.tubes[data.relationships.tube.data.id]
-        delete this.resources.tubes[data.relationships.tube.data.id]
 
         //Populate libraries
         this.resources.libraries = dataToObjectById({
           data: libraries,
           includeRelationships: true,
         })
 
-        //Assign library request to tube if the tube has a library
-        Object.values(this.resources.libraries).forEach((library) => {
-          const request = this.resources.requests[library.request]
-          this.resources.tubes[library.tube].requests = [request.id]
-          this.resources.tubes[library.tube].source_id = String(library.id)
+        // populate tubes and assign libraries to tubes
+        this.resources.tubes = assignLibraryRequestsToTubes({
+          libraries: this.resources.libraries,
+          requests: this.resources.requests,
+          tubes,
         })
+
+        // Get the pool tube and remove it from tubes list.
+        // There must be a reason why it is done this way?
+        this.tube = this.resources.tubes[data.relationships.tube.data.id]
+        delete this.resources.tubes[data.relationships.tube.data.id]
```

Comment:
I might be wrong, but is it that we are removing the pool tube so that it does not appear in the library tube list in places it is used, maybe  the pool edit page?

Reply:
That was my suspicion.

In which case I think the best course of action would be to rejig the store so that we can add separation for the different types of tubes rather than adding them to the resources and then removing them.

Reply:
Yes, agree. I don't think the store data should be tweaked for that purpose.